cli/src/commands/operation/log.rs

```diff
@@ -190,18 +196,29 @@ fn do_op_log(
     let mut formatter = ui.stdout_formatter();
     let formatter = formatter.as_mut();
     let limit = args.limit.unwrap_or(usize::MAX);
-    let iter = op_walk::walk_ancestors(slice::from_ref(current_op)).take(limit);
+    let iter = op_walk::walk_ancestors(slice::from_ref(current_op));
+
     if !args.no_graph {
         let mut raw_output = formatter.raw()?;
         let mut graph = get_graphlog(graph_style, raw_output.as_mut());
-        for op in iter {
+        let iter = iter.map(|op| -> Result<_, OpStoreError> {
             let op = op?;
-            let mut edges = vec![];
-            for id in op.parent_ids() {
-                edges.push(Edge::Direct(id.clone()));
-            }
+            let edges = op.parents().map_ok(GraphEdge::direct).try_collect()?;
```

Comment:
Loading `op.parents()` here isn't really required. I'm wondering if we should update `reverse_graph` to add an `id_fn`? Although we will still end up having to load most operations in most cases, so perhaps it's not worth optimizing.

Reply:
Yeah, maybe that's good idea, but I also agree that the extra I/O wouldn't matter in practice.